### PR TITLE
remove strcpy

### PR DESCRIPTION
**Pull request by rgetz**
This is a follow on from #439 

It slowly updates some of the core files in a way that should be easier to review, and follow, and bisect. Since this effects the way that the internal xml is built up, one file at a time is changed, the downsides is there are 10 commits in the PR.

- It tracks the length of building up xml files
- adds a iio_strlcpy
- uses things, so there is no chance of undetected buffer overflow
- optimize things to use memcpy where possible.

I will apply the other changes that were in the old branch if/when this gets accepted. That would removes these: 
- 7 instances of `snprintf` (should be using `iio_snprintf`)
- 2 instances of `sprintf`(should be using `iio_snprintf`)
- 8 instances of `strncpy` (should be using `iio_strlcpy`)
- 1 instances of `strcpy` (should be using `iio_strlcpy`)

### CODE REVIEW

File: channel.c [hunk 1]
```diff
@@ -289,6 +298,11 @@ char * iio_channel_get_xml(const struct iio_channel *chn, size_t *length)
 
 	strcpy(ptr, "</channel>");
 	*length = ptr - str + sizeof("</channel>") - 1;
+	len -= sizeof("</channel>");
+
+	if (len < 0)
+		IIO_ERROR("Internal libIIO error: iio_channel_get_xml str length isssue\n");
```
[the remark this answers is not in the document]
Comment (rgetz) [reply channel.c:311]: OK - updated in all files.

File: iio-private.h [hunk 2]
```diff
@@ -65,6 +65,10 @@
 #define MAX_CHN_NAME 256 /* encoded in the sysfs filename */
 #define MAX_DEV_ID 256 /* encoded in the sysfs filename */
 #define MAX_DEV_NAME 256 /* encoded in the sysfs filename */
+#define MAX_CTX_NAME 256    /* nominally "xml" */
+#define MAX_CTX_DESC 256    /* nominally "linux ..." */
+#define MAX_ATTR_NAME 256   /*encoded in the sysfs filename */
```
[the remark this answers is not in the document]
Comment (rgetz) [reply iio-private.h]: fixed.

**Comment by rgetz** [comment]
OK - updated with feedback.

**Comment by dNechita** [comment]
I will merge this in the next 12-18 hours if nobody objects until then.

**Pull request merged**